src/Components/Components/test/ParameterViewTest.cs

```diff
@@ -402,6 +402,172 @@ public void Clone_ParameterPreservesOrder()
+            RenderTreeFrame.Attribute(3, "attribute 3", "value 3"),
+        }, 0);
+        var newParameters = new ParameterView(ParameterViewLifetime.Unbound, new[]
+        {
+            RenderTreeFrame.Element(0, "some element").WithElementSubtreeLength(3),
+            RenderTreeFrame.Attribute(1, "attribute 1", "value 1"),
+            RenderTreeFrame.Attribute(2, "attribute 3", "value 3"),
+        }, 0);
+
+        // Act
+        var hasRemovedDirectParameters = newParameters.HasRemovedDirectParameters(oldParameters);
+
+        // Assert
+        Assert.True(hasRemovedDirectParameters);
+    }
+
+    [Fact]
+    public void HasRemovedDirectParameters_ParameterReplaced()
+    {
+        // Arrange
+        var oldParameters = new ParameterView(ParameterViewLifetime.Unbound, new[]
+        {
+            RenderTreeFrame.Element(0, "some element").WithElementSubtreeLength(4),
+            RenderTreeFrame.Attribute(1, "attribute 1", "value 1"),
+            RenderTreeFrame.Attribute(2, "attribute 2", "value 2"),
+            RenderTreeFrame.Attribute(3, "attribute 3", "value 3"),
+        }, 0);
+        var newParameters = new ParameterView(ParameterViewLifetime.Unbound, new[]
+        {
+            RenderTreeFrame.Element(0, "some element").WithElementSubtreeLength(4),
+            RenderTreeFrame.Attribute(1, "attribute 2", "value 1"),
+            RenderTreeFrame.Attribute(2, "attribute replaced", "value 2"),
+            RenderTreeFrame.Attribute(3, "attribute 3", "value 3"),
+        }, 0);
+
+        // Act
+        var hasRemovedDirectParameters = newParameters.HasRemovedDirectParameters(oldParameters);
+
+        // Assert
+        Assert.True(hasRemovedDirectParameters);
+    }
+
+    [Fact]
+    public void HasRemovedDirectParameters_ParameterReplacedAndAdded()
+    {
+        // Arrange
+        var oldParameters = new ParameterView(ParameterViewLifetime.Unbound, new[]
+        {
+            RenderTreeFrame.Element(0, "some element").WithElementSubtreeLength(4),
+            RenderTreeFrame.Attribute(1, "attribute 1", "value 1"),
+            RenderTreeFrame.Attribute(2, "attribute 2", "value 2"),
+            RenderTreeFrame.Attribute(3, "attribute 3", "value 3"),
+        }, 0);
+        var newParameters = new ParameterView(ParameterViewLifetime.Unbound, new[]
+        {
+            RenderTreeFrame.Element(0, "some element").WithElementSubtreeLength(5),
+            RenderTreeFrame.Attribute(1, "attribute 2", "value 1"),
+            RenderTreeFrame.Attribute(2, "attribute replaced", "value 2"),
+            RenderTreeFrame.Attribute(3, "attribute 3", "value 3"),
+            RenderTreeFrame.Attribute(4, "attribute 4", "value 3"),
+        }, 0);
+
+        // Act
+        var hasRemovedDirectParameters = newParameters.HasRemovedDirectParameters(oldParameters);
+
+        // Assert
+        Assert.True(hasRemovedDirectParameters);
+    }
+
+    [Fact]
+    public void HasRemovedDirectParameters_ParametersSwapped()
```

Comment:
What happens if you have multiple parameters with the same name that are swapped?

Reply:
In that case, `HasRemovedDirectParameters()` returns `false`, but the component would still update as it would for any other hot reload update (we just wouldn't treat it as a parameter removal, which is fine because no parameter is getting removed).